x-pack/plugins/code/server/security.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import Boom from 'boom';
+import { Request, ResponseToolkit, Server, ServerRoute } from 'hapi';
+
+export interface SecuredRoute extends ServerRoute {
+  requireRoles?: [string];
+  requireAdmin?: boolean;
+}
+
+export const ADMIN_ROLE = 'code_admin';
```

Comment:
Do we need to define this role in Elasticsearch's list of reserved roles?
I think we've tried to avoid hard-coding roles in this fashion within Kibana. It might be worthwhile to revisit this once we merge [Granular App Privileges](https://github.com/elastic/kibana/issues/20277). Since Code is entirely within Kibana, and not an ES Cluster app, we can probably get away with defining your permissions using Kibana's privilege model. 

/cc @kobelb